src/Controls/tests/DeviceTests/Elements/TextInput/TextInputTests.cs

```diff
@@ -1,24 +1,338 @@
-﻿using Microsoft.Maui.Controls;
-using Microsoft.Maui.Handlers;
-using Microsoft.Maui.Hosting;
+﻿using System.Threading.Tasks;
+using Microsoft.Maui.Controls;
+using Xunit;
 
 namespace Microsoft.Maui.DeviceTests
 {
-
 	[Category(TestCategory.TextInput)]
-	public partial class TextInputTests : ControlsHandlerTestBase
+	public abstract partial class TextInputTests<THandler, TView> : ControlsHandlerTestBase
+		where THandler : class, IViewHandler, IPlatformViewHandler, new()
+		where TView : InputView, IView, ITextInput, new()
 	{
-		void SetupBuilder()
+		protected abstract int GetPlatformCursorPosition(THandler handler);
+
+		protected abstract int GetPlatformSelectionLength(THandler handler);
+
+		protected abstract Task<string> GetPlatformText(THandler handler);
+
+		[Theory(DisplayName = "Text is Transformed Correctly at Initialization")]
+		[ClassData(typeof(TextTransformCases))]
+		public async Task InitialTextTransformApplied(string text, TextTransform transform, string expected)
+		{
+			var control = new TView() { Text = text, TextTransform = transform };
+			var platformText = await GetPlatformText(await CreateHandlerAsync<THandler>(control));
+			Assert.Equal(expected, platformText);
+		}
+
+		[Theory(DisplayName = "Text is Transformed Correctly after Initialization")]
+		[ClassData(typeof(TextTransformCases))]
+		public async Task TextTransformUpdated(string text, TextTransform transform, string expected)
+		{
+			var control = new TView() { Text = text };
+			var handler = await CreateHandlerAsync<THandler>(control);
+			await InvokeOnMainThreadAsync(() => control.TextTransform = transform);
+			var platformText = await GetPlatformText(handler);
+			Assert.Equal(expected, platformText);
+		}
+
+		[Theory(DisplayName = "Unset CursorPosition is kept at zero at initialization with TextTransform")]
+		[InlineData("This is a test!!!")]
+		[InlineData("a")]
+		[InlineData("")]
+		[InlineData(" ")]
+		public async Task UnsetCursorPositionKeepsToZeroOnInitializationWithTextTransform(string text)
+		{
+			var editor = new TView
```

Comment:
Use `control` as above or `textInput`
not an `Editor`, so it can be a bit confusing

```suggestion
			var control = new TView
```

Reply:
same comment for all "entry" and "editor" vars for all the tests in this file